pkg/infinispan/cluster.go

```diff
@@ -80,8 +80,7 @@ func (c Cluster) GetClusterSize(podName string) (int, error) {
 
 // GracefulShutdown performs clean cluster shutdown
 func (c Cluster) GracefulShutdown(podName string) error {
-	// TODO can this be HEAD?
-	_, err, reason := c.Client.Get(podName, consts.ServerHTTPClusterStop, nil)
+	_, err, reason := c.Client.Post(podName, consts.ServerHTTPClusterStop, "", nil)
```

Comment:
@ryanemerson have you validated the upgrade path?
For 2.0.x -> master due to the #581 issue Infinispan image never been updated, but for 1.1.x it could handle an error, an old image may not support a new HTTP verb

Reply:
> @ryanemerson have you validated the upgrade path? The scripts are ready, you can try to use, but I'm waiting on

No. I have never looked at these tests tbh and don't know how they work. Do we have any docs/notes anywhere?

> For 2.0.x -> master due to the #581 issue Infinispan image never been updated, but for 1.1.x it could handle an error, an old image may not support the new HTTP verb

POST actions for graceful shutdown were added in Infinispan 11 [ISPN-11967](https://issues.redhat.com/browse/ISPN-11967) and the corresponding GET methods removed in 12. So it should be possible to upgrade from the `2.0.x` operator to `master`, e.g. Infinispan 11->12

Reply:
Yes, @ryanemerson , we should be create a backward compatible operator code, as anyone can customize image or use another version rather that default
Just two ideas:
- Use CLI inside container instead of REST call for graceful shutdown only. This needs to be verified
- Invoke with POST verb and if this call returns 405 error code try to call with GET

Reply:
There has to be a limit on what Infinispan versions the operator can manage, we can't maintain backwards compatibility indefinitely as it's not flexible enough. @rigazilla What guarantees do we provide for upgrading? e.g. 10.1.x -> 12.x or just 11.x -> 12.x? We need to determine this, then we can implement a suitable solution.

Reply:
I think we can suppose upgrades to be incremental: i.e. 10.x -> 11.x -> 12.x.
Operator catalog works that way, so I don't think we need to support 10.x -> 12.x.

Reply:
This PR should be Ok to merge then